### DISCUSSION TITLE
MIDI learn support

### DISCUSSION THREAD

**Discussion by e-b** [FigBug/Wavetable#46] [Ideas]
The synth is so direct and quick to use, it really asks to be mapped to a multi-knob controller like a MIDI Fighter Twister or a Faderfox. MIDI learn would just be amazing!

**Comment by FigBug** [comment]
A good idea, are there any synths you use that do this well from UI perspective?

**Comment by e-b** [comment]
Yes, there is kind of a common pattern that is very intuitive among several synths - you click a MIDI button in the UI header or menu (to activate MIDI learn), then all mappable controls are highlighted, then you turn the hardware know / move the slider you want to map and move the UI control to be mapped at the same time. When you are done you toggle the MIDI learn off again.

**Reply by FigBug** [reply]
And then that is saved for all instances for the plugin? Or different instances can hav different midi maps?

**Comment by e-b** [comment]
For Phase Plant that I used a lot, just one setting is saved for all
instances of that session.
It would be very powerful to just save and restore settings to and from a
named file.
